dll/win32/msgina/msgina.c

```diff
@@ -172,94 +168,63 @@ GetRegistrySettings(PGINA_CONTEXT pgContext)
         return FALSE;
     }
 
-    rc = ReadRegSzValue(hKey,
-                        L"AutoAdminLogon",
-                        &lpAutoAdminLogon);
+    rc = ReadRegDwordValue(hKey, L"AutoAdminLogon", &dwValue);
     if (rc == ERROR_SUCCESS)
-    {
-        if (wcscmp(lpAutoAdminLogon, L"1") == 0)
-            pgContext->bAutoAdminLogon = TRUE;
-    }
-
+        pgContext->bAutoAdminLogon = !!dwValue;
     TRACE("bAutoAdminLogon: %s\n", pgContext->bAutoAdminLogon ? "TRUE" : "FALSE");
 
-    rc = ReadRegDwordValue(hKey,
-                           L"DisableCAD",
-                           &dwDisableCAD);
+    // TODO: What to do also depends on whether we are on Terminal Services.
+    rc = ReadRegDwordValue(hKey, L"DisableCAD", &dwDisableCAD);
     if (rc == ERROR_SUCCESS)
     {
         if (dwDisableCAD != 0)
             pgContext->bDisableCAD = TRUE;
     }
-
     TRACE("bDisableCAD: %s\n", pgContext->bDisableCAD ? "TRUE" : "FALSE");
 
+    // TODO: The default value is always read from the registry.
+    // By default: Workstation: TRUE; Server: FALSE.
+    // However: Set it to TRUE always on SafeMode. Keep it FALSE
+    // on Sessions != 0 Terminal Services.
```

Comment:
This is hardly readable, the sentence is backwards and formatted for a phone screen?